src/main.tsx

```diff
@@ -620,6 +659,8 @@ function main() {
         </React.StrictMode>
     )
 
+    checkAndMigrateBooksIDsMap()
```

Comment:
@brunojm i think you want to block on this until the migration is finished right? you don't want the sync starting until the migration is done?

do you have access to `await` here? if so you could just `await checkAndMigrateBooksIDsMap()`

otherwise, you could use `then`s